SwiftReflector/OverrideBuilder.cs

```diff
@@ -191,6 +199,32 @@ ExtensionDeclaration BuildExtensionDefinition (ModuleDeclaration targetModule)
 			return decl;
 		}
 
+		ClassDeclaration BuildAssociatedTypeOverride (string name, ModuleDeclaration targetModule)
+		{
+			name = name ?? ProxyClassName (OriginalClass);
+			var protocol = OriginalClass as ProtocolDeclaration;
```

Comment:
You are doing and as, yet not check for protocol to be null. We might have issues.